pydantic_ai_slim/pydantic_ai/profiles/anthropic.py

```diff
@@ -1,8 +1,57 @@
 from __future__ import annotations as _annotations
 
+from copy import deepcopy
+from dataclasses import dataclass
+
+from .._json_schema import JsonSchema, JsonSchemaTransformer
 from . import ModelProfile
 
 
+@dataclass(init=False)
+class AnthropicJsonSchemaTransformer(JsonSchemaTransformer):
+    """Transforms schemas to the subset supported by Anthropic structured outputs.
+
+    Anthropic's SDK `transform_schema()` automatically:
+    - Adds `additionalProperties: false` to all objects (required by API)
+    - Removes unsupported constraints (minLength, pattern, etc.)
+    - Moves removed constraints to description field
+    - Removes title and $schema fields
+
+    When `strict=None`, we compare before/after to detect if constraints were dropped.
+    """
+
+    def walk(self) -> JsonSchema:
+        from anthropic import transform_schema
+
+        schema = super().walk()
+
+        if self.strict is False:
+            # no transformation if specifically non-strict
+            return schema
+
+        if self.strict is None:
+            before = deepcopy(schema)
+            transformed = transform_schema(schema)
+            if before != transformed:
+                self.is_strict_compatible = False
+            return transformed
```

Comment:
Shouldn't we return the unmodified `before` if `self.is_strict_compatible is False`?

Reply:
I think this is a worthwhile discussion:

anthropic's `transform_schema` changes the schema in the way that's detailed above in the docstring

that doesn't seem too lossy to me, so _my_ opinion is we should allow the transform when `self.strict is None`

so, not even making the lossy check, because in that case, setting the `is_strict_compatible = False`

what do you opine?